src/CMakeLists.txt

```diff
@@ -13,6 +13,15 @@ project(ImpactT LANGUAGES Fortran)
 
 option(USE_MPI "Activate MPI build" OFF)
 
+# Allow type mismatch for GCC 10
+# (Adapted from from: https://github.com/Unidata/netcdf-fortran/blob/main/CMakeLists.txt)
+# Check to see if compiler supports
+# -fallow-argument-mismatch-flag introduced in gcc 10.
+include(CheckFortranCompilerFlag)
+check_fortran_compiler_flag("-fallow-argument-mismatch" COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
+if(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
+  set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fallow-argument-mismatch")
+endif(COMPILER_HAS_ALLOW_ARGUMENT_MISMATCH)
```

Comment:
Please move this down into the `if(USE_MPI)` block and use:
```
target_compile_options(${OUTPUT_NAME} PUBLIC "-fallow-argument-mismatch")
```

Reply:
done

Reply:
Thanks! Does the no-MPI version compile without `-fallow-argument-mismatch` on GCC 10 for you?

Reply:
Yes, works fine, but many warnings: https://github.com/impact-lbl/IMPACT-T/issues/20